packages/react/block.ts

```diff
@@ -65,17 +65,14 @@ export const block = <P extends MillionProps>(
 
     const childrenSize = portalRef.current.length;
     const children = new Array(childrenSize);
-
-    children[0] = marker;
-    children[1] = createElement(Effect, {
-      effect,
-      deps: hmrTimestamp ? [hmrTimestamp] : [],
-    });
     for (let i = 0; i < childrenSize; ++i) {
-      children[i + 2] = portalRef.current[i]?.portal;
+      children[i] = portalRef.current[i]?.portal;
     }
 
-    const vnode = createElement(Fragment, { children });
+    const vnode = createElement(Fragment, {}, marker, createElement(Effect, {
+      effect,
+      deps: hmrTimestamp ? [hmrTimestamp] : [],
+    }), children);
```

Comment:
is it possible that portals (children) won't have a key and the warning would be shown again?

What I'd suggest is that you spread the children by `, ...children`, so this way the whole error would be removed. 

```
}), ...children);
```

Reply:
spreading might hit the arguments bound because it has a lower bound than arrays.